src/llmcompressor/transformers/finetune/data/data_args.py

```diff
@@ -31,26 +33,41 @@ class CustomDataTrainingArguments(DVCDatasetTrainingArguments):
         },
     )
 
-    text_column: Optional[str] = field(
+    text_column: str = field(
         default="text",
-        metadata={"help": "For custom datasets only. The text field key"},
+        metadata={
+            "help": (
+                "Optional key to be used as the `text` input to tokenizer/processor "
+                "after dataset preprocesssing"
+            )
+        },
     )
 
     remove_columns: Union[None, str, List] = field(
         default=None,
-        metadata={"help": "Column names to remove after preprocessing custom datasets"},
+        metadata={"help": "Column names to remove after preprocessing (depreciated)"},
     )
 
     preprocessing_func: Union[None, str, Callable] = field(
         default=None,
         metadata={
             "help": (
-                "The preprocessing function to apply or the preprocessing func name in "
-                "src/llmcompressor/transformers/utils/preprocessing_functions.py"
+                "Typically a function which applies a chat template. Can take the form "
+                "of iither a function to apply to the dataset, a name defined in "
+                "src/llmcompressor/transformers/utils/preprocessing_functions.py, or "
+                "a path to a function definition of the form /path/to/file.py:func"
             )
         },
     )
 
+    data_collator: Callable[[Any], Any] = field(
+        default_factory=lambda: DefaultDataCollator(),
+        metadata={
+            "help": "For custom datasets only. The function to used to form a batch "
+            "from the dataset"
```

Comment:
Is it specifically to form the batches within sample sets i.e it isn't needed if we are doing batch size=1 with multiple samples?

Reply:
The `DefaultDataCollator` fails to correctly collate vision datasets, batch_size=1 or otherwise. This is because of weirdness around how some processors will automatically batch the data they're given, or how some models with interpret lists of pixel_values as multiple images for a single batch, rather than multiple single-image batches.

For that reason, we must expose this argument to override the `DefaultDataCollator`